addons/cookoff/functions/fnc_detonateAmmunition.sqf

```diff
@@ -27,21 +31,22 @@ private _magazine = _magazines select _magazineIndex;
 _magazine params ["_magazineClassname", "_amountOfMagazines"];
 
 if (_amountOfMagazines > 0) exitWith {
-    private _newMagCount = _amountOfMagazines - floor(1 + random(6));
-    if (_newMagCount <= 0) then {
+    private _removed = _amountOfMagazines min floor(1 + random(6));
+
+    _amountOfMagazines = _amountOfMagazines - _removed;
+    if (_amountOfMagazines <= 0) then {
         _magazines deleteAt _magazineIndex;
     } else {
-        _magazine set [1, _newMagCount]; // clear out the magazine
+        _magazine set [1, _amountOfMagazines]; // clear out the magazine
     };
-    private _timeBetweenAmmoDetonation = (random 7) * (1 / random (_amountOfMagazines)) min MAX_TIME_BETWEEN_AMMO_DET;
-    _timeBetweenAmmoDetonation = _timeBetweenAmmoDetonation max 0.1;
+    private _timeBetweenAmmoDetonation = (((random 10) / (sqrt _totalAmmo)) min MAX_TIME_BETWEEN_AMMO_DET) max 0.1;
```

Comment:
:+1: Excellent improvement, the inverse proportionality of average delay to the square root of remaining ammo count is a very elegant solution to simulating the decaying intensity of the burn.

Should drastically reduce the length of cook off now that there's far lass chance of the max time being hit with each delay.